### PR TITLE
added the Query capability to the list items/wishlist endpoints

### PR DESCRIPTION
**Pull request by wyam1**

### CODE REVIEW

**Comment by wyam1** [comment]
Added some code here without the test cases. Hoping for one of you @aklufas or @jku8 to take a look at the code and then we can start writing the test cases together.

**Comment by aklufas** [comment]
@jku8 can you take a look at this? the test I was trying to write doesn't work and I'm stuck 

test_routes.py > lines 180ish
query_wishlists_by_name

**Comment by jku8** [comment]
@aklufas @wyam1 you also need to modify the routes.py file for the /wishlists and /items endpoint to handle the query logic. You can do this with an if, else.

**Comment by aklufas** [comment]
Updated, struggling with creating a create_items() helper function.

File: tests/test_routes.py [hunk 1]
```diff
@@ -67,12 +67,15 @@ def _create_wishlists(self, count):
     
     def _create_items(self, count):
         """ Factory method to create items in bulk """
+        wl = WishList()
```
Comment (jku8) [inline tests/test_routes.py]: you need to create a wishlist with some row data: wl = WishList(name='wishlist1', category=category1')

Comment (aklufas) [reply tests/test_routes.py]: Ok - I changed that. 

I'm getting stuck on item.create() in the for loop. This is the error that keeps coming up, I've tried googling it but am completely stuck.

Screen Shot 2021-03-31 at 3 02 51 PM

Comment (aklufas) [reply tests/test_routes.py]: Actually, I think I resolved this error by adding a wishlist id. Now, I'm getting a foreign key error. 
Screen Shot 2021-03-31 at 3 30 12 PM

**Comment by aklufas** [comment]
@jku8 - finished the tests, this is ready for review!

**Comment by ajaffery08** [comment]
LGTM

**Pull request merged**